deftest/telescope.lua

```diff
@@ -332,6 +332,48 @@ function table.reverse(t)
      end
 end
 
+-- Setup a new environment suitable for running a new test
+local function newEnv()
+  local env = {}
+  
+  -- Make sure globals are accessible in the new environment
+  setmetatable(env, {__index = _G})
+
+  -- Setup all the assert functions in the new environment
+  for k, v in pairs(assertions) do
+    setfenv(v, env)
+    env[k] = v
+  end
+
+  return env
+end
+
+-- Invoke test inside the environment.
+local function invoke_test(func, env)
+  local assertions_invoked = 0
+  env.assertion_callback = function()
+    assertions_invoked = assertions_invoked + 1
+  end
+  setfenv(func, env)
+  local result, message = xpcall(func, debug.traceback)
+  if result and assertions_invoked > 0 then
+    return status_codes.pass, assertions_invoked, nil
+  elseif result then
+    return status_codes.unassertive, 0, nil
+  elseif type(message) == "table" then
+    return status_codes.fail, assertions_invoked, message
+  else
+    return status_codes.err, assertions_invoked, {message, debug.traceback()}
+  end
+end
+
+-- Invoke test without adding it to the report.
+-- This function is used for invoking integration tests
+-- to get predictable results.
```

Comment:
What do you mean by predictable results?

Reply:
It is unfortunate that telescope.lua has to be modified since it makes upgrading it (if a new version is released) so much harder.

Reply:
Thanks for the reply! I execute integration tests separately, because if I just add them with other tests to telescope, the state of the game world can change when telescope runs them. That's why I execute tests right after the event that triggered them happened and add results to telescope.
I agree modifying telescope.lua is not the best solution. I will think how to do it differently.

Reply:
Ah, I see. But how about running the integration tests separately? Integration tests usually take longer to complete and because of this you may not want to run them together with the unit tests (which should complete very quickly).

Reply:
Agree